File: app/frontend/shared/domain/trigger-repository.ts
import { Trigger, TriggerDiscordServerId, TriggerId } from './trigger'

export interface TriggerRepository {
  findById(uuid: TriggerId): Promise<Trigger>
  findByDiscordServerId(discordServerId: TriggerDiscordServerId): Promise<Trigger[]>
  save(trigger: Trigger): Promise<void>
  delete(trigger: Trigger): Promise<void>
}

File: context/src/main/kotlin/com/usadapekora/context/domain/trigger/TriggerRepository.kt
package com.usadapekora.context.domain.trigger

interface TriggerRepository {
    fun all(): Array<Trigger>
    fun find(id: Trigger.TriggerId): Trigger
    fun findByDiscordServer(id: Trigger.TriggerDiscordGuildId): Array<Trigger>
    fun save(entity: Trigger)
    fun delete(entity: Trigger)
}
